source/extensions/common/wasm/ext/envoy_proxy_wasm_api.h

```diff
@@ -43,3 +60,55 @@ inline std::vector<DnsResult> parseDnsResults(StringView data) {
   }
   return results;
 }
+
+inline StatResult parseStatResults(StringView data) {
+  StatResult results;
+  uint32_t data_len = 0;
+  while (data_len < data.length()) {
+    const uint32_t* n = reinterpret_cast<const uint32_t*>(data.data() + data_len);
+    uint32_t block_size = *n++;
+    uint32_t block_type = *n++;
+    uint32_t num_stats = *n++;
+    if (block_type == 1) { // counter
```

Comment:
Maybe use `enum` for those known types?

Reply:
converting to a StatType enum in the next revision